Lib/ensurepip/__init__.py

```diff
@@ -167,24 +141,20 @@ def _bootstrap(*, root=None, upgrade=False, user=False,
     with tempfile.TemporaryDirectory() as tmpdir:
         # Put our bundled wheels into a temporary directory and construct the
         # additional paths that need added to sys.path
-        additional_paths = []
-        for name, package in _get_packages().items():
-            if package.wheel_name:
-                # Use bundled wheel package
-                wheel_name = package.wheel_name
-                wheel_path = resources.files("ensurepip") / "_bundled" / wheel_name
-                whl = wheel_path.read_bytes()
-            else:
-                # Use the wheel package directory
-                with open(package.wheel_path, "rb") as fp:
-                    whl = fp.read()
-                wheel_name = os.path.basename(package.wheel_path)
-
-            filename = os.path.join(tmpdir, wheel_name)
-            with open(filename, "wb") as fp:
-                fp.write(whl)
-
-            additional_paths.append(filename)
+        package = _get_pip_info()
+        wheel_name = package["filename"]
+        if package["bundled"]:
+            # Use bundled wheel package
+            wheel_path = resources.files("ensurepip") / "_bundled" / wheel_name
+            whl = wheel_path.read_bytes()
+        else:
+            # Use the wheel package directory
+            with open(os.path.join(_WHEEL_PKG_DIR, wheel_name), "rb") as fp:
+                whl = fp.read()
+
+        filename = os.path.join(tmpdir, wheel_name)
+        with open(filename, "wb") as fp:
+            fp.write(whl)
```

Comment:
If you change to using ``pathlib``, we could leverage the traversable API a little here:

```suggestion
        if package["bundled"]:
            # Use bundled wheel package
            wheel_path = resources.files("ensurepip") / "_bundled" / wheel_name
        else:
            # Use the wheel package directory
            wheel_path  = Path(_WHEEL_PKG_DIR, wheel_name)

        whl = wheel_path.read_bytes()
        filename = Path(tmpdir, wheel_name)
        filename .write_bytes(whl)
```

Reply:
I think that's what I've been doing in the original PR, just with a `nullcontext` for clever code linearization...
https://github.com/python/cpython/commit/c31bdca823eab7e4ffcd70c7bd9a9895506e1e33#diff-f5e321563d5bc98371a960662f654a9387fd24f9bacb382d5b90d417ad2569c7R238

Reply:
https://github.com/python/cpython/commit/b2a69a121a82d5abc8b99eed2492c4602d9fb91b implemented this in a more linear way. I'm marking the thread as resolved.